BraveRewardsUI/Common/DisclaimerLinkLabel.swift

```diff
@@ -0,0 +1,134 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+
+/// A Disclaimer Label that allows clickable links (or data-detectors)
+final class DisclaimerLinkLabel: UITextView {
+  
+  /// Called when a link is tapped
+  var onLinkedTapped: ((Link) -> Void)?
+  
+  /// Types of links supported by this label
+  public enum Link {
+    case unknown
+    case termsOfService
+    case privacyPolicy
+  }
+  
+  override var text: String? {
+    didSet {
+      updateText()
+    }
+  }
+  
+  /// Converts the text into attributed text for display
+  func updateText() {
+    guard let text = text else {
+      super.text = nil
+      return
+    }
+    
+    let attributedString = { () -> NSAttributedString? in
+      let paragraphStyle = NSMutableParagraphStyle()
+      paragraphStyle.alignment = .center
+      
+      let attributes: [NSAttributedString.Key: Any] = [.font: self.font ?? UIFont.systemFont(ofSize: 12.0),
+                                                       .foregroundColor: self.textColor ?? UX.textColor,
+                                                       .paragraphStyle: paragraphStyle]
+      
+      let text = try? NSMutableAttributedString(data: text.data(using: .utf8) ?? Data(),
+                                                          options: [.documentType: NSAttributedString.DocumentType.html,
```

Comment:
Is this still performant using HTML parsing over just specifying `.link` attributes manually?

Reply:
I haven't noticed any performance issues since the text is very small and we're setting it only once on creation of the labels. In a tableview it might be a problem.